### PR TITLE
Add support for setting company logo

### PR DESCRIPTION
**Pull request by richtier**
[this task](https://uktrade.atlassian.net/browse/ED-188)

We can now simply pass the `logo` file to the api client and it will upload the file alongside the rest of the form data.

### CODE REVIEW

File: exportdirectory/base.py [hunk 1]
```diff
@@ -21,26 +21,39 @@ def __init__(self, base_url=None, api_key=None):
         self.api_key = api_key
 
     def put(self, url, data):
-        return self.request("PUT", url, "application/json", data=data)
+        return self.request(
+            "PUT", url, "application/json", data=json.dumps(data)
```
Comment (konradko) [inline exportdirectory/base.py]: can you please use kwargs for all?

File: exportdirectory/base.py [hunk 2]
```diff
@@ -21,26 +21,39 @@ def __init__(self, base_url=None, api_key=None):
...
+        )
 
-    def patch(self, url, data):
-        return self.request("PATCH", url, "application/json", data=data)
+    def patch(self, url, data, files={}):
```
Comment (konradko) [inline exportdirectory/base.py]: do not use mutable types as default value, bad things can happen

File: exportdirectory/base.py [hunk 3]
```diff
@@ -21,26 +21,39 @@ def __init__(self, base_url=None, api_key=None):
...
+        if files:
+            response = self.request(
+                "PATCH", url, "multipart/form-data", data=data, files=files
+            )
+        else:
+            response = self.request(
+                "PATCH", url, "application/json", data=json.dumps(data),
+            )
+        return response
```
Comment (konradko) [inline exportdirectory/base.py]: this flow is fine, maybe just DRY up with sth like 

```
def patch(self, url, data, **kwargs):
    if kwargs.get('files'):
        content_type = "multipart/form-data"
    else:
        content_type = "application/json"

    return self.request(
        "PATCH", url, content_type, data=json.dumps(data), **kwargs
    )
```

File: exportdirectory/base.py [hunk 4]
```diff
@@ -21,26 +21,39 @@ def __init__(self, base_url=None, api_key=None):
...
 
     def get(self, url, params=None):
         return self.request("GET", url, params=params)
 
     def post(self, url, data):
-        return self.request("POST", url, "application/json", data=data)
+        return self.request(
+            "POST", url, "application/json", data=json.dumps(data)
```
Comment (konradko) [inline exportdirectory/base.py]: looks like "application/json" could be put as a constant somewhere. Can we maybe also import it from ``requests``?

File: exportdirectory/base.py [hunk 5]
```diff
@@ -21,26 +21,39 @@ def __init__(self, base_url=None, api_key=None):
...
+        )
 
     def delete(self, url, data=None):
         return self.request("DELETE", url)
 
-    def request(self, method, url, content_type=None, data=None, params=None):
+    def request(
+        self, method, url, content_type=None, data=None, params=None,
```
Comment (konradko) [inline exportdirectory/base.py]: given that nothing happens to these vars, maybe just ``**kwargs``?

File: tests/test_company.py [hunk 6]
```diff
@@ -13,10 +14,33 @@ def setUp(self):
...
     @stub_request('https://example.com/company/1/', 'patch')
-    def test_update_profile(self, stub):
+    def test_update_profile_handles_data(self, stub):
         data = {'key': 'value'}
         self.client.update_profile(id=1, data=data)
-        assert stub.request_history[0].json() == data
+        request = stub.request_history[0]
+        assert request.json() == data
+
+    @stub_request('https://example.com/company/1/', 'patch')
```
Comment (konradko) [inline tests/test_company.py]: so nice

File: exportdirectory/base.py (hunk 3 above)
Comment (richtier) [inline exportdirectory/base.py]: notice were also doing `json.dumps` only in one flow.

sure, will leave as is for now.

Comment (konradko) [inline exportdirectory/base.py]: yeah so:

```
def patch(self, url, data, **kwargs):
    if kwargs.get('files'):
        content_type = "multipart/form-data"
    else:
        content_type = "application/json"
        data = json.dumps(data)

    return self.request(
        "PATCH", url, content_type, data, **kwargs
    )
```

File: exportdirectory/base.py (hunk 4 above)
Comment (richtier) [inline exportdirectory/base.py]: It's not available in requests lib to import. I will do a follow up pr to make some constants.

File: exportdirectory/base.py (hunk 5 above)
Comment (richtier) [inline exportdirectory/base.py]: we would then be doing `kwargs['url']`, and `kwargs['content_type']` when we use those variables, while feels a bit abusive. I find it useful seeing the function signature explicitly.

**Pull request merged**